File: tests/fixtures/kafkajs/basic/no_imports.ts
// File without kafkajs import — must produce zero emits even though
// it happens to have a .send({ topic }) shape.

interface FakeProducer {
  send(opts: { topic: string }): Promise<void>;
}

const fake = { async send(_opts: { topic: string }) {} } as FakeProducer;

export async function localSend() {
  await fake.send({ topic: 'not-a-kafka-topic' });
}

File: tests/fixtures/kafkars/basic/src/no_imports.rs
// File without rdkafka use — must produce zero emits even though it
// has a `subscribe` method and a `::to` style call.

pub struct FakeBus;

impl FakeBus {
    pub fn subscribe(&self, _topics: &[&str]) {}
}

pub fn local(b: &FakeBus) {
    b.subscribe(&["not-a-kafka-topic"]);
}
